web/pgadmin/static/js/components/ReactCodeMirror/index.jsx

```diff
@@ -59,30 +63,45 @@ CopyButton.propTypes = {
 };
 
 
-export default function CodeMirror({className, currEditor, showCopyBtn=false, customKeyMap=[], onTextSelect, ...props}) {
+export default function CodeMirror({className, currEditor, showCopyBtn=false, customKeyMap=[], onTextSelect, eventBus, ...props}) {
   const editor = useRef();
   const [[showFind, isReplace, findKey], setShowFind] = useState([false, false, false]);
   const [showGoto, setShowGoto] = useState(false);
   const [showCopy, setShowCopy] = useState(false);
-
+  const preferences = usePreferences().getPreferencesForModule('sqleditor');
   const finalCustomKeyMap = useMemo(()=>[{
-    key: 'Mod-f', run: () => {
+    key: toCodeMirrorKey(preferences.find), run: () => {
       setShowFind(prevVal => [true, false, !prevVal[2]]);
     },
     preventDefault: true,
     stopPropagation: true,
   }, {
-    key: 'Mod-Alt-f', run: () => {
+    key: toCodeMirrorKey(preferences.replace), run: () => {
       setShowFind(prevVal => [true, true, !prevVal[2]]);
     },
     preventDefault: true,
     stopPropagation: true,
   }, {
-    key: 'Mod-l', run: () => {
+    key: toCodeMirrorKey(preferences.goto_line_col), run: () => {
       setShowGoto(true);
     },
     preventDefault: true,
     stopPropagation: true,
+  }, {
+    key: toCodeMirrorKey(preferences.comment), run: () => {
+      editor.current?.execCommand('toggleComment');
+    },
+    preventDefault: true,
+    stopPropagation: true,
+  },{
+    key: toCodeMirrorKey(preferences.format_sql), run: () => {
```

Comment:
this should not come here. Format SQL is not codemirror internal.
Or - you can add new command for formatSql just like toggleComment or others - because format here will work only is query tool and not outside.

Reply:
If we press the shortcut for format sql outside it will not cause any error or issues that's why I kept here. I will add a separate command